### DISCUSSION TITLE
Bug with Stage Manager

### DISCUSSION THREAD

**Discussion by weicheng998** [chrenn/swish-dl#18] [Q&A]
I recently encountered a bug when using the centering feature with the stage manager.

Whenever Swish is launched after the stage manager (i.e. launch Swich when stage manager is already on), the centering gesture will not position the window correctly in the middle. Instead, it will shift the window a little bit to the right. 

Before centering:
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/c309e55a-62c2-4686-8493-6948e328bef6">

After centering:
<img width="1727" alt="image" src="https://github.com/user-attachments/assets/e2c1f8e8-4e98-4eb4-a8c4-cab6e9ea5e6a">

Here's my setting for the stage manager:
<img width="477" alt="image" src="https://github.com/user-attachments/assets/93e4113e-bdd1-492f-8d7d-7a1dfed3400e">
(I noticed the 'show recent app' option might alter the behavior)

I'm on Macbook Pro 16 with M2 Pro, MacOS 14.5, Swish 1.10.3.

Currently, I use the following shortcut to solve the issue:
<img width="757" alt="image" src="https://github.com/user-attachments/assets/ab0631ac-ae3c-4e79-9196-a036dcaaef18">

PS. I'm not a software developer but I notice the stage manager may affect the windowing of a few apps. From my perception, Apple reserves some space on the left to 'show recent app' so the desktop area is either shrunk or shifted to the right.

**Comment by chrenn** [comment]
That's working as intended! I don't use Stage Manager myself, but isn't having that little strip the whole point?

Anyway, Swish 2 (paid upgrade coming later this year) will feature more Stage Manager customization.
